### PR TITLE
Add junit test case for SQL API.

### PR DESCRIPTION
**Pull request by kul**
Opens #55

### CODE REVIEW

**Comment by kul** [comment]
https://travis-ci.org/databricks/spark-csv/jobs/60802910#L1201

**Comment by falaki** [comment]
I believe to be able to persist a table, you need a HiveContext. In the unit tests, we create a SQLContext. For the unit test take a look at ```org.apache.spark.sql.hive.test.TestHiveContext```

Related to #55 make sure you use HiveContext (in addition to using Spark 1.3.x)

**Comment by kul** [comment]
@falaki  Sorry i do not understand this. In this very repo for Scala test suite `CsvSuite` you have the exact same test cases working on plain sql context not hive context?

```scala
  test("DDL test") {
    sql(
      s"""
        |CREATE TEMPORARY TABLE carsTable
        |USING com.databricks.spark.csv
        |OPTIONS (path "$carsFile", header "true")
      """.stripMargin.replaceAll("\n", " "))

    assert(sql("SELECT year FROM carsTable").collect().size === numCars)
  }
```

**Comment by falaki** [comment]
That test is creating TEMPORARY table. The schema for temp table is kept in memory and is not persisted to (hive) metastore.

**Comment by kul** [comment]
Didnt you say to use temporary only before 1.3.x support? It was failing for temporary as well.

**Comment by kul** [comment]
https://travis-ci.org/databricks/spark-csv/jobs/60807732#L1259

File: src/test/java/com/databricks/spark/csv/JavaCsvSuite.java [hunk 1]
```diff
@@ -39,6 +39,15 @@ public void testCsvParser() {
   }
 
   @Test
+  public void testSQL() {
+      sqlContext.sql("CREATE TEMPORARY TABLE carsTable " +
+                      "USING com.databricks.spark.csv " +
+                      "OPTIONS (path \""+carsFile+"\", header \"true\"");
```
Comment (falaki) [inline src/test/java/com/databricks/spark/csv/JavaCsvSuite.java]: You are missing a closing parenthesis here. It should be:
```java
sqlContext.sql("CREATE TEMPORARY TABLE carsTable " +
                      "USING com.databricks.spark.csv " +
                      "OPTIONS (path \""+carsFile+"\", header \"true\")");
```

**Comment by kul** [comment]
aa

Thanks! Sorry for wasting your time.

**Pull request closed without merging**